### PR TITLE
Ignore variance annotations

### PR DESCRIPTION
**Pull request by gasche**

### CODE REVIEW

File: miniml/compiler/compile.scm [hunk 1]
```diff
@@ -151,9 +151,20 @@
 
    (type_name_with_args
     (LIDENT) : $1
-    (QUOTE LIDENT LIDENT) : $3
+    (variance QUOTE LIDENT LIDENT) : $3
```
Comment (Ekdohibs) [inline miniml/compiler/compile.scm:154]: Shouldn't this be `$4`?

File: miniml/compiler/compile.scm [hunk 2]
```diff
@@ -151,9 +151,20 @@
...
 
+   (variance
+    (nonempty_variance) : '()
+    ( ) : '()
+   )
+
+   (nonempty_variance
+    (PLUS) : '()
+    (MINUS) : '()
+    (EQ) : '()
```
Comment (Ekdohibs) [inline miniml/compiler/compile.scm:165]: Since when is `=` a variance? I don't see it in the manual.

**Review by Ekdohibs — COMMENTED** [review]
Seems good! Just a few slight changes to make before merging.

File: miniml/compiler/compile.scm (hunk 2 above)
Comment (gasche) [reply miniml/compiler/compile.scm:165]: Apparently it exists in my brain only, as an explicit way to specify "invariant".

File: miniml/compiler/compile.scm (hunk 1 above)
Comment (gasche) [reply miniml/compiler/compile.scm]: fixed, thanks

**Comment by gasche** [comment]
Note: 4.11 introduced injectivity annotations `!'a` (can also be combined with variances), but I would wait until we encounter one to do the work of adding a `!` token to the grammar.

**Comment by Ekdohibs** [comment]
Thanks!

**Pull request merged**